### PR TITLE
KMS integration tests

### PR DESCRIPTION
**Pull request by nico-stefani**
|Related issue|
|-------------|
|      #3339       |

## Description

This PR adds tests for AWS KMS integration

#### Tier 0
- Default configuration tests
- `remove_from_bucket` parameter tests
- `only_logs_after` parameter tests
- `path` parameter tests
- `discard_regex` parameter tests

#### Tier 1
- `only_logs_after` parameter tests
---

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @nico-stefani  (Developer)  | `tests/integration/test_aws` | ⚫⚫⚫ | :green_circle: :green_circle:  | Ubuntu 22.04    |      | Nothing to highlight |
| @fdalmaup  (Reviewer)   | | ⚫⚫⚫ | :green_circle: :green_circle: |  Ubuntu 20.04  |      | Nothing to highlight |

### Tier 0 results
```python
root@ubuntu-jammy:/home/vagrant/qa/tests/integration/test_aws# pytest -k kms --tier 0
================================================================================================ test session starts ================================================================================================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/vagrant/qa/tests/integration, configfile: pytest.ini
plugins: metadata-2.0.2, html-3.1.1, testinfra-5.0.0
collected 24 items / 16 deselected / 8 selected

test_basic.py .                                                                                                                                                                                               [ 12%]
test_discard_regex.py .                                                                                                                                                                                       [ 25%]
test_only_logs_after.py ..                                                                                                                                                                                    [ 50%]
test_path.py ...                                                                                                                                                                                              [ 87%]
test_remove_from_bucket.py .                                                                                                                                                                                  [100%]

================================================================================================= warnings summary ==================================================================================================
../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:433
  /usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:433: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/nodeids
    config.cache.set("cache/nodeids", sorted(self.cached_nodeids))

../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/stepwise.py:52
  /usr/local/lib/python3.10/dist-packages/_pytest/stepwise.py:52: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/stepwise
    session.config.cache.set(STEPWISE_CACHE_DIR, [])

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================= 8 passed, 16 deselected, 2 warnings in 161.84s (0:02:41) ==============================================================================
```

### Tier 1 results

```python
root@ubuntu-jammy:/home/vagrant/qa/tests/integration/test_aws# pytest -k kms --tier 1
================================================================================================ test session starts ================================================================================================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/vagrant/qa/tests/integration, configfile: pytest.ini
plugins: metadata-2.0.2, html-3.1.1, testinfra-5.0.0
collected 24 items / 23 deselected / 1 selected

test_only_logs_after.py .                                                                                                                                                                                     [100%]

================================================================================================= warnings summary ==================================================================================================
../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:433
  /usr/local/lib/python3.10/dist-packages/_pytest/cacheprovider.py:433: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/nodeids
    config.cache.set("cache/nodeids", sorted(self.cached_nodeids))

../../../../../../usr/local/lib/python3.10/dist-packages/_pytest/stepwise.py:52
  /usr/local/lib/python3.10/dist-packages/_pytest/stepwise.py:52: PytestCacheWarning: cache could not write path /home/vagrant/qa/tests/integration/.pytest_cache/v/cache/stepwise
    session.config.cache.set(STEPWISE_CACHE_DIR, [])

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=================================================================================== 1 passed, 23 deselected, 2 warnings in 32.76s ===================================================================================
```

### CODE REVIEW

**Review by fdalmaup — CHANGES_REQUESTED** [review]
The Tier 1 test gave the following error
<details><summary>Error logs</summary>

    root@ubuntu-focal:/home/vagrant/wazuh-qa/tests/integration/test_aws# pytest -k kms --tier 1
    ====================================================================== test session starts ======================================================================
    platform linux -- Python 3.8.10, pytest-5.0.0, py-1.11.0, pluggy-0.13.1
    rootdir: /home/vagrant/wazuh-qa/tests/integration, inifile: pytest.ini
    plugins: html-2.0.1, metadata-2.0.4, testinfra-5.0.0
    collected 24 items / 23 deselected / 1 selected                                                                                                                 
    
    test_only_logs_after.py F                                                                                                                                 [100%]
    
    =========================================================================== FAILURES ============================================================================
    ____________________________________________________ test_multiple_calls[kms_only_logs_after_multiple_calls] ____________________________________________________
    
    metadata = {'bucket_name': 'wazuh-kms-integration-tests', 'bucket_type': 'custom', 'description': 'KMS only_logs_after multiple calls configurations', 'name': 'kms_only_logs_after_multiple_calls'}
    clean_s3_cloudtrail_db = None, load_wazuh_basic_configuration = None, restart_wazuh_function = None, delete_file_from_s3 = None
    
        @pytest.mark.tier(level=1)
        @pytest.mark.parametrize('metadata', t3_configuration_metadata, ids=t3_case_ids)
        def test_multiple_calls(
            metadata, clean_s3_cloudtrail_db, load_wazuh_basic_configuration, restart_wazuh_function, delete_file_from_s3
        ):
            """
            description: Call the AWS module multiple times with different only_logs_after values.
            test_phases:
                - setup:
                    - Delete the `s3_cloudtrail.db`.
        
                - test:
                    - Call the module without only_logs_after and check that no logs were processed.
                    - Upload a log file for the day of the test execution and call the module with the same parameters as
                      before, check that the uploaded logs were processed.
                    - Call the module with the same parameters and check that no logs were processed, there were no duplicates.
                    - Call the module with only_logs_after set in the past and check that the expected number of logs were
                      processed.
                    - Call the module with the same parameters in and check there were no duplicates.
                    - Call the module with only_logs_after set with an older date check that old logs were processed without
                      duplicates.
                    - Call the module with only_logs_after set with an early date than setted previously and check that no logs
                      were processed, there were no duplicates.
        
                - teardown:
                    - Delete the `s3_cloudtrail.db`.
                    - Delete the uploaded files.
            wazuh_min_version: 4.5.0
            parameters:
                - metadata:
                    type: dict
                    brief: Get metadata from the module.
                - clean_s3_cloudtrail_db:
                    type: fixture
                    brief: Delete the DB file before and after the test execution.
                - load_wazuh_basic_configuration:
                    type: fixture
                    brief: Load basic wazuh configuration.
                - restart_wazuh_daemon:
                    type: fixture
                    brief: Restart the wazuh service.
                - delete_file_from_s3:
                    type: fixture
                    brief: Delete the a file after the test execution.
            input_description:
                - The `cases_multiple_calls` file provides the test cases.
            """
            ONLY_LOGS_AFTER_PARAM = '--only_logs_after'
        
            bucket_type = metadata['bucket_type']
            bucket_name = metadata['bucket_name']
        
            base_parameters = [
                '--bucket', bucket_name,
                '--type', bucket_type,
                '--regions', 'us-east-1',
                '--aws_profile', 'qa',
                '--debug', '2'
            ]
        
            # Call the module without only_logs_after and check that no logs were processed
            last_marker_key = datetime.now().strftime(cons.PATH_DATE_FORMAT)
            if bucket_type == cons.CUSTOM_TYPE:
                event_monitor.check_marker_from_output(
                    command_output=call_aws_module(*base_parameters),
                    file_key=last_marker_key
                )
            else:
                event_monitor.check_non_processed_logs_from_output(
                    command_output=call_aws_module(*base_parameters),
                    bucket_type=bucket_type
                )
        
            # Call the module with only_logs_after set in the past and check that the expected number of logs were
            # processed
            event_monitor.check_processed_logs_from_output(
                command_output=call_aws_module(*base_parameters, ONLY_LOGS_AFTER_PARAM, '2022-NOV-20'),
                expected_results=3
            )
        
            # Call the module with the same parameters in and check there were no duplicates
    >       event_monitor.check_non_processed_logs_from_output(
                command_output=call_aws_module(*base_parameters, ONLY_LOGS_AFTER_PARAM, '2022-NOV-20'),
                bucket_type=bucket_type
            )
    
    test_only_logs_after.py:347: 
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
    /usr/local/lib/python3.8/dist-packages/wazuh_testing/modules/aws/event_monitor.py:91: in check_non_processed_logs_from_output
        analyze_command_output(
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
    
    command_output = 'DEBUG: +++ Debug mode on - Level: 2\nDEBUG: Generating default configuration for retries: mode standard - max_attempt... file: 2022/11/26/firehose_kms-1-20221126T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json\nDEBUG: +++ DB Maintenance\n'
    callback = <function make_aws_callback.<locals>.<lambda> at 0x7fab6d337040>, expected_results = 1, error_message = 'Some logs may were processed'
    
        def analyze_command_output(
            command_output: str, callback: Callable = _default_callback, expected_results: int = 1, error_message: str = ''
        ):
            """Analyze the given command output searching for a pattern.
        
            Args:
                command_output (str): The output to analyze.
                callback (Callable, optional): A callback to process each line. Defaults to _default_callback.
                expected_results (int, optional): Number of expected results. Defaults to 1.
                error_message (str, optional): Message to show with the exception. Defaults to ''.
        
            Raises:
                OutputAnalysisError: When the expected results are not correct.
            """
        
            results = []
        
            for line in command_output.splitlines():
                logger.debug(line)
                item = callback(line)
        
                if item is not None:
                    results.append(item)
        
            results_len = len(results)
        
            if results_len != expected_results:
                if error_message:
                    logger.error(error_message)
                    logger.error('Results found: %s', results_len)
                    logger.error('Results expected: %s', expected_results)
    >               raise OutputAnalysisError(error_message)
    E               wazuh_testing.modules.aws.cli_utils.OutputAnalysisError: Some logs may were processed
    
    /usr/local/lib/python3.8/dist-packages/wazuh_testing/modules/aws/cli_utils.py:65: OutputAnalysisError
    --------------------------------------------------------------------- Captured stderr call ----------------------------------------------------------------------
    2023-01-24 19:23:36,401 - wazuh_testing - DEBUG - Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-kms-integration-tests', '--type', 'custom', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2']'
    2023-01-24 19:23:38,718 - wazuh_testing - DEBUG - DEBUG: +++ Debug mode on - Level: 2
    2023-01-24 19:23:38,718 - wazuh_testing - DEBUG - DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
    2023-01-24 19:23:38,718 - wazuh_testing - DEBUG - DEBUG: +++ Table does not exist; create
    2023-01-24 19:23:38,718 - wazuh_testing - DEBUG - DEBUG: +++ Marker: 2023/01/24
    2023-01-24 19:23:38,718 - wazuh_testing - DEBUG - DEBUG: +++ DB Maintenance
    2023-01-24 19:23:38,718 - wazuh_testing - DEBUG - Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-kms-integration-tests', '--type', 'custom', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2', '--only_logs_after', '2022-NOV-20']'
    2023-01-24 19:23:41,897 - wazuh_testing - DEBUG - DEBUG: +++ Debug mode on - Level: 2
    2023-01-24 19:23:41,898 - wazuh_testing - DEBUG - DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
    2023-01-24 19:23:41,898 - wazuh_testing - DEBUG - DEBUG: +++ Marker: 2022/11/20
    2023-01-24 19:23:41,898 - wazuh_testing - DEBUG - DEBUG: ++ Found new log: 2022/11/21/firehose_kms-1-20221121T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json
    2023-01-24 19:23:41,898 - wazuh_testing - DEBUG - DEBUG: ++ Found new log: 2022/11/23/firehose_kms-1-20221123T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json
    2023-01-24 19:23:41,898 - wazuh_testing - DEBUG - DEBUG: ++ Found new log: 2022/11/26/firehose_kms-1-20221126T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json
    2023-01-24 19:23:41,898 - wazuh_testing - DEBUG - DEBUG: +++ DB Maintenance
    2023-01-24 19:23:41,898 - wazuh_testing - DEBUG - Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-kms-integration-tests', '--type', 'custom', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2', '--only_logs_after', '2022-NOV-20']'
    2023-01-24 19:23:44,012 - wazuh_testing - DEBUG - DEBUG: +++ Debug mode on - Level: 2
    2023-01-24 19:23:44,012 - wazuh_testing - DEBUG - DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
    2023-01-24 19:23:44,012 - wazuh_testing - DEBUG - DEBUG: +++ Marker: 2022/11/20
    2023-01-24 19:23:44,012 - wazuh_testing - DEBUG - DEBUG: ++ Skipping previously processed file: 2022/11/21/firehose_kms-1-20221121T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json
    2023-01-24 19:23:44,012 - wazuh_testing - DEBUG - DEBUG: ++ Skipping previously processed file: 2022/11/23/firehose_kms-1-20221123T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json
    2023-01-24 19:23:44,012 - wazuh_testing - DEBUG - DEBUG: ++ Skipping previously processed file: 2022/11/26/firehose_kms-1-20221126T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json
    2023-01-24 19:23:44,012 - wazuh_testing - DEBUG - DEBUG: +++ DB Maintenance
    2023-01-24 19:23:44,012 - wazuh_testing - ERROR - Some logs may were processed
    2023-01-24 19:23:44,012 - wazuh_testing - ERROR - Results found: 3
    2023-01-24 19:23:44,013 - wazuh_testing - ERROR - Results expected: 1
    ----------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------
    DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-kms-integration-tests', '--type', 'custom', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2']'
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Table does not exist; create
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: 2023/01/24
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
    DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-kms-integration-tests', '--type', 'custom', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2', '--only_logs_after', '2022-NOV-20']'
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: 2022/11/20
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: 2022/11/21/firehose_kms-1-20221121T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: 2022/11/23/firehose_kms-1-20221123T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Found new log: 2022/11/26/firehose_kms-1-20221126T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
    DEBUG    wazuh_testing:cli_utils.py:23 Calling AWS module with: '[PosixPath('/var/ossec/wodles/aws/aws-s3'), '--bucket', 'wazuh-kms-integration-tests', '--type', 'custom', '--regions', 'us-east-1', '--aws_profile', 'qa', '--debug', '2', '--only_logs_after', '2022-NOV-20']'
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Debug mode on - Level: 2
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: Generating default configuration for retries: mode standard - max_attempts 10
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ Marker: 2022/11/20
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: 2022/11/21/firehose_kms-1-20221121T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: 2022/11/23/firehose_kms-1-20221123T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: ++ Skipping previously processed file: 2022/11/26/firehose_kms-1-20221126T1324Z_1f396a2d-7e70-4b9c-8005-ba0389ebbd60.json
    DEBUG    wazuh_testing:cli_utils.py:52 DEBUG: +++ DB Maintenance
    ERROR    wazuh_testing:cli_utils.py:62 Some logs may were processed
    ERROR    wazuh_testing:cli_utils.py:63 Results found: 3
    ERROR    wazuh_testing:cli_utils.py:64 Results expected: 1
    =========================================================== 1 failed, 23 deselected in 24.83 seconds ============================================================


</details>

**Comment by nico-stefani** [comment]
> The Tier 1 test gave the following error
> Error logs

Fixed in a08e52bde7dda31b87ddb9c5f592c1edf869eecf

File: deps/wazuh_testing/wazuh_testing/modules/aws/event_monitor.py [hunk 1]
```diff
@@ -76,8 +82,11 @@ def check_processed_logs_from_output(command_output: str, expected_results: int
     )
 
 
-def check_non_processed_logs_from_output(command_output: str, expected_results: int = 1):
-    pattern = r'.*DEBUG: \+\+\+ No logs to process in bucket: '
+def check_non_processed_logs_from_output(command_output: str, bucket_type: str, expected_results: int = 1):
```
Comment (fdalmaup) [inline deps/wazuh_testing/wazuh_testing/modules/aws/event_monitor.py:85]: Docstring is missing

File: deps/wazuh_testing/wazuh_testing/modules/aws/event_monitor.py [hunk 2]
```diff
@@ -62,6 +63,11 @@ def callback_detect_event_processed(line):
         return line
 
 
+def callback_detect_event_processed_or_skipped(pattern):
```
Comment (fdalmaup) [inline deps/wazuh_testing/wazuh_testing/modules/aws/event_monitor.py:66]: Docstring is missing

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py [hunk 3]
```diff
@@ -94,22 +103,150 @@ def get_data_sample(self) -> dict:
...
+        now = datetime.now()
+        path = f'{self.BASE_PATH}{now.strftime(cons.PATH_DATE_FORMAT)}/'
+        name = (
+            f'{self.BASE_FILE_NAME}{cons.FLOW_LOG_ID}_{now.strftime(cons.FILENAME_DATE_FORMAT)}_{abs(hash(now))}'
+            f'{cons.LOG_EXT}'
+        )
+
+        return f'{path}{name}'
+
+    def get_data_sample(self) -> str:
```
Comment (fdalmaup) [inline deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:126]: Docstring is missing

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py [hunk 4]
```diff
@@ -1,11 +1,20 @@
...
+from io import StringIO
 from os.path import join
 from uuid import uuid4
 
+from wazuh_testing.tools.utils import get_random_ip, get_random_port, get_random_string
+
 from . import constants as cons
 
 
+def get_random_interface_id():
```
Comment (fdalmaup) [inline deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:14]:
```suggestion
def get_random_interface_id() -> str:
```
Also, a docstring should be included

**Review by fdalmaup — CHANGES_REQUESTED** [review]

File: deps/wazuh_testing/wazuh_testing/modules/aws/event_monitor.py (hunk 1 above)
Comment (nico-stefani) [reply deps/wazuh_testing/wazuh_testing/modules/aws/event_monitor.py:147]: added in 11868d3ed5858c2f1c0f4f63bd254d45766c0565

File: deps/wazuh_testing/wazuh_testing/modules/aws/event_monitor.py (hunk 2 above)
Comment (nico-stefani) [reply deps/wazuh_testing/wazuh_testing/modules/aws/event_monitor.py]: added in 11868d3ed5858c2f1c0f4f63bd254d45766c0565

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py (hunk 3 above)
Comment (nico-stefani) [reply deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:126]: Please disregard this, already has been fixed in https://github.com/wazuh/wazuh-qa/pull/3699/.

File: deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py [hunk 5]
```diff
@@ -413,24 +413,113 @@ def get_data_sample(self) -> str:
...
+            str: Synthetic filename.
+        """
+        now = datetime.utcnow()
+        path = join(self.BASE_PATH, now.strftime(cons.PATH_DATE_FORMAT))
+        name = f"{self.BASE_FILE_NAME}{now.strftime(cons.FILENAME_DATE_FORMAT)}_{str(uuid4())}{cons.JSON_EXT}"
+
+        return join(path, name)
+
+    def get_data_sample(self) -> str:
+        """Returns a sample of data according to the KMS format.
```
Comment (davidjiglesias) [inline deps/wazuh_testing/wazuh_testing/modules/aws/data_generator.py:436]:
```suggestion
        """Return a sample of data according to the KMS format.
```

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**